terraform/eval_diff.go

```diff
@@ -238,6 +238,38 @@ func (n *EvalDiffTainted) Eval(ctx EvalContext) (interface{}, error) {
 	return nil, nil
 }
 
+// EvalFilterDiff is an EvalNode implementation that filters the diff
+// according to some filter.
```

Comment:
This is named to sound like a generic filter, but for that we'd need it to take a function, yeah?

This is really `EvalDestroyStrippingFilter` at this point, if I'm understanding properly.